app/services/reports/reports_types_creator.rb

```diff
@@ -0,0 +1,178 @@
+module Reports
+  class ReportsTypesCreator # rubocop:disable Metrics/ClassLength
```

Comment:
This is a big class 😬  There's probably too much going on it.

Would it be possible to split some of it out into another class to handle searching (ie moving the `search_applications` method and everything it calls from this class into the new one) which would just leave this one to actually create the report.

I haven't tried it so I'm not sure how easy it would be to do, but maybe worth considering.

Reply:
That was possibly something I was going to do later. Just didn't want to spend lots of time doing all that if a) we break it up later if design, UR have some input, b) it didn't get merged because people didn't like it lol.

Reply:
I could also point your attention to `app/models/legal_aid_application.rb` in the meantime 😬 😂

Reply:
Fair point. Don't spend any time on it now. Let's get some views on the new functionality and then we can worry about refactoring.

Reply:
> I could also point your attention to `app/models/legal_aid_application.rb` in the meantime as a form of distraction 😬 😂

Good point, well made